Apps/WebClient/src/NewClientApp/src/components/private/timeline/comment/CommentComponent.vue

```diff
@@ -0,0 +1,153 @@
+
+interface Props {
+    comment: UserComment;
+}
+const commentComponentProps = defineProps<Props>();
+
+const logger = container.get<ILogger>(SERVICE_IDENTIFIER.Logger);
+const userStore = useUserStore();
+const commentStore = useCommentStore();
+const appStore = useAppStore();
+
+const commentInput = ref("");
+const isEditMode = ref(false);
+const isUpdating = ref(false);
+
+const isMobile = computed(() => appStore.isMobile);
+
+function formatDate(date: string): string {
+    return new DateWrapper(date, { isUtc: true }).format("yyyy-MMM-dd, t");
+}
+
+function onCancel(): void {
+    isEditMode.value = false;
+}
+
+function editComment(): void {
+    commentInput.value = commentComponentProps.comment.text;
+    isEditMode.value = true;
+}
+
+function handleCommentError(err: Error): void {
+    logger.error(JSON.stringify(err));
+    window.scrollTo({ top: 0, behavior: "smooth" });
+}
+
+function onSubmit(): void {
+    isUpdating.value = true;
+    const comment: UserComment = {
+        id: commentComponentProps.comment.id,
+        text: commentInput.value,
+        userProfileId: commentComponentProps.comment.userProfileId,
+        parentEntryId: commentComponentProps.comment.parentEntryId,
+        createdDateTime: commentComponentProps.comment.createdDateTime,
+        entryTypeCode: commentComponentProps.comment.entryTypeCode,
+        version: commentComponentProps.comment.version,
+    };
+    commentStore
+        .updateComment(userStore.user.hdid, comment)
+        .then(() => {
+            logger.info("Comment Updated");
+            isEditMode.value = false;
+        })
+        .catch(handleCommentError)
+        .finally(() => (isUpdating.value = false));
+}
+
+function removeComment(): void {
+    if (confirm("Are you sure you want to delete this comment?")) {
+        isUpdating.value = true;
+        commentStore
+            .deleteComment(userStore.user.hdid, commentComponentProps.comment)
+            .then(() => logger.info("Comment removed"))
+            .catch(handleCommentError)
+            .finally(() => (isUpdating.value = false));
+    }
+}
+</script>
+
+<template>
+    <v-sheet class="px-4 py-2 my-2" color="grey-lighten-5">
+        <v-row v-if="!isEditMode">
+            <v-col>
+                <p data-testid="commentText" class="text-body-1">
+                    {{ comment.text }}
+                    <br />
+                    <span class="text-caption text-medium-emphasis">
+                        {{ formatDate(comment.createdDateTime) }}
+                    </span>
+                </p>
+            </v-col>
+            <v-col cols="auto">
+                <v-menu data-testid="commentMenuBtn">
+                    <template #activator="{ props }">
+                        <HgIconButtonComponent
+                            icon="ellipsis-v"
+                            size="small"
+                            v-bind="props"
+                        />
+                    </template>
```

Comment:
To avoid naming conflicts, rather than renaming the component's props, it's much cleaner to rename these:

```suggestion
                    <template #activator="{ props: slotProps }">
                        <HgIconButtonComponent
                            icon="ellipsis-v"
                            size="small"
                            v-bind="slotProps"
                        />
                    </template>
```

Reply:
Thanks, I saw this recently on one of your PR's, I've adopted this since then 👍